src/ScreenshotManager.vala

```diff
@@ -342,19 +342,40 @@ namespace Gala {
             }
         }
 
-        private static bool save_image_to_clipboard (Cairo.ImageSurface image, string filename, out string used_filename) {
+        private bool save_image_to_clipboard (Cairo.ImageSurface image, string filename, out string used_filename) {
             used_filename = filename;
 
-            unowned Gdk.Display display = Gdk.Display.get_default ();
-            unowned Gtk.Clipboard clipboard = Gtk.Clipboard.get_default (display);
-
             var screenshot = Gdk.pixbuf_get_from_surface (image, 0, 0, image.get_width (), image.get_height ());
             if (screenshot == null) {
-                warning ("could not save screenshot to clipboard: null pixbuf");
+                warning ("Could not save screenshot to clipboard: null pixbuf");
+                return false;
+            }
+
+            var stream = new GLib.MemoryOutputStream.resizable ();
```

Comment:
I'm really not to knowledgeable in this area so what's the advantage of using a memory stream here? Wouldn't `save_to_buffer` together with `new Bytes.take ()` do the same? Or is saving to a memory stream here somehow faster?

Reply:
I used it because gnome-shell uses it, `GLib.Bytes.take` works just fine. Thank you.